File: examples/gas-calc/thermo-curves-O2/thermo-curves-for-O2.lua
-- A script to output Cp and h for O2 over temperature range 200--20000 K.
--
-- Author: Rowan J. Gollan
-- Date: 2016-12-23
--
-- To run this script:
-- $ prep-gas O2.inp O2-gas-model.lua
-- $ gas-calc thermo-curves-for-O2.lua
--

gasModelFile = 'O2-gas-model.lua'
gmodel = GasModel:new{gasModelFile}

gs = GasState:new{gmodel}
gs.p = 1.0e5 -- Pa
gs.massf = {O2=1.0}

outputFile = 'O2-thermo.dat'
print("Opening file for writing: ", outputFile)
f = assert(io.open(outputFile, "w"))
f:write("#  1:T[K]      2:Cp[J/kg/K]     3:h[J/kg]\n")

Tlow = 200.0
Thigh = 20000.0
dT = 100.0

for T=Tlow,Thigh,dT do
   gs.T = T
   gmodel:updateThermoFromPT(gs)
   Cp = gmodel:Cp(gs)
   h = gmodel:enthalpy(gs)
   f:write(string.format(" %12.6e %12.6e %12.6e\n", T, Cp, h))
end

f:close()
print("File closed. Done.")

File: examples/gas-calc/thermo-curves-O2/thermo-curves-for-O2.py
# A script to output Cp and h for O2 over temperature range 200--20000 K.
#
# Author: Peter J. and Rowan J. Gollan
# Date: 2019-11-21
#
# To run this script:
# $ prep-gas O2.inp O2-gas-model.lua
# $ python3 thermo-curves-for-O2.py
#
from gdtk.gas import GasModel, GasState

gasModelFile = 'O2-gas-model.lua'
gmodel = GasModel(gasModelFile)

gs = GasState(gmodel)
gs.p = 1.0e5 # Pa
gs.massf = {"O2":1.0}

outputFile = 'O2-thermo.dat'
print("Opening file for writing: %s" % outputFile)
f = open(outputFile, "w")
f.write("#  1:T[K]      2:Cp[J/kg/K]     3:h[J/kg]\n")

lowT = 200.0
dT = 100.0

for i in range(199):
    gs.T = dT*i + lowT
    gs.update_thermo_from_pT()
    f.write(" %12.6e %12.6e %12.6e\n" % (gs.T, gs.Cp, gs.enthalpy))

f.close()
print("File closed. Done.")
